openedx/adg/lms/applications/models.py

```diff
@@ -139,21 +138,21 @@ class UserApplication(TimeStampedModel):
     cover_letter = models.TextField(blank=True, verbose_name=_('Cover Letter'), )
 
     OPEN = 'open'
-    ACCEPTED = 'accepted'
     WAITLIST = 'waitlist'
+    ACCEPTED = 'accepted'
 
     STATUS_CHOICES = (
         (OPEN, _('Open')),
-        (ACCEPTED, _('Accepted')),
         (WAITLIST, _('Waitlist')),
+        (ACCEPTED, _('Accepted'))
     )
     status = models.CharField(
         verbose_name=_('Application Status'), choices=STATUS_CHOICES, max_length=8, default=OPEN,
     )
     reviewed_by = models.ForeignKey(
         User, null=True, blank=True, on_delete=models.CASCADE, verbose_name=_('Reviewed By')
     )
-    internal_admin_note = models.TextField(null=True, blank=True, verbose_name=_('Admin Note'))
+    internal_admin_note = models.TextField(blank=True, verbose_name=_('Admin Note'))
```

Comment:
@javeriaejaz-arbisoft Django does not recommend using `null=True` for CharField or TextField. The following excerpt is taken from [Django official docs](https://docs.djangoproject.com/en/3.1/ref/models/fields/#null):

"Avoid using null on string-based fields such as CharField and TextField. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” the Django convention is to use the empty string, not NULL."

As far as `blank=True` is concerned, if I remove it, the field will be made mandatory and any QA while creating a test user application from admin will have to first add an internal note, which should not be the case. The QA needs to create user application with blank note, so they can later populate the note and run test cases related to adding notes while reviewing an application.

As far as assigning a default value is concerned, by default, the default value for a TextField or CharField is an empty string. But yes, it may cause a problem while running a migration on dev/staging/prod server if there are existing applications with internal_admin_note = null, and we might have to cater that. The problem occurred locally for me as well and I had to provide a default '' while running the migration in the command line. Well pointed out! For that, instead of remaking the migration with default='', a better approach would be to make sure that before running deployment in dev/staging/prod, there are no user application with empty internal note. We can manually populate that, since ADG isn't live yet. There are not too many applications on those servers. Only three were there in dev, and one in staging. I have updated them to avoid this issue. Explicitly providing an empty string as default may not be very favourable, because as I mentioned it is already defined for TextField in Django.